src/util/object_forwarding.rs

```diff
@@ -0,0 +1,212 @@
+use crate::util::metadata::MetadataSpec;
```

Comment:
I think a layer like this is actually good. Its abstraction level is between the general interface in `VMObjectModel` and each single use of the metadata in mmtk-core everywhere. It provides specific APIs for each spec. We could have one for each spec (or each spec that is complicate enough).